timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/AllNodesDisabledNamespacesUpdater.java

```diff
@@ -151,64 +171,80 @@ private boolean isSuccessfulOnAllNodes(Function<DisabledNamespacesUpdaterService
 
-        return attemptOnAllNodes(namespaces, lockId, update, localUpdate);
+        return attemptOnAllNodes(namespaces, lockId, update, localUpdate, true);
+    }
+
+    private Collection<SingleNodeUpdateResponse> reEnableNamespacesOnRemoteNodes(
+            ReenableNamespacesRequest request, List<DisabledNamespacesUpdaterService> nodes) {
+        Function<DisabledNamespacesUpdaterService, SingleNodeUpdateResponse> update =
+                node -> node.reenable(authHeader, request);
+
+        Map<DisabledNamespacesUpdaterService, CheckedRejectionExecutorService> correspondingExecutors =
+                KeyedStream.stream(remoteExecutors).filterKeys(nodes::contains).collectToMap();
+        PaxosResponsesWithRemote<DisabledNamespacesUpdaterService, SingleNodeUpdateResponse> responses =
+                PaxosQuorumChecker.collectAllResponses(
+                        ImmutableList.copyOf(nodes), update, correspondingExecutors, Duration.ofSeconds(5L), false);
+        return responses.responses().values();
     }
 
     // Update and analysis
-    private List<SingleNodeUpdateResponse> attemptOnAllNodes(
+    private AllNodesUpdateResponse attemptOnAllNodes(
             Set<Namespace> namespaces,
             UUID lockId,
             Function<DisabledNamespacesUpdaterService, SingleNodeUpdateResponse> update,
-            Supplier<SingleNodeUpdateResponse> localUpdate) {
-        List<SingleNodeUpdateResponse> responses = attemptOnAllRemoteNodes(update);
-        SingleNodeUpdateResponse localResponse = updateLocallyOrCheckState(responses, namespaces, lockId, localUpdate);
-        responses.add(localResponse);
-        return responses;
+            Supplier<SingleNodeUpdateResponse> localUpdate,
+            boolean alwaysAttemptOnLocalNode) {
+        Map<DisabledNamespacesUpdaterService, SingleNodeUpdateResponse> remoteResponses =
+                attemptOnAllRemoteNodes(update);
+        SingleNodeUpdateResponse localResponse = updateLocallyOrCheckState(
+                remoteResponses.values(), namespaces, lockId, localUpdate, alwaysAttemptOnLocalNode);
+        return AllNodesUpdateResponse.builder()
+                .remoteResponses(remoteResponses)
+                .localResponse(localResponse)
+                .build();
     }
 
     private SingleNodeUpdateResponse updateLocallyOrCheckState(
-            List<SingleNodeUpdateResponse> responses,
+            Collection<SingleNodeUpdateResponse> responses,
             Set<Namespace> namespaces,
             UUID lockId,
-            Supplier<SingleNodeUpdateResponse> localUpdate) {
-        if (responses.stream().allMatch(SingleNodeUpdateResponse::getWasSuccessful)
-                && responses.size() == remoteUpdaters.size()) {
+            Supplier<SingleNodeUpdateResponse> localUpdate,
+            boolean alwaysAttemptOnLocalNode) {
+        if (alwaysAttemptOnLocalNode
+                || (responses.stream().allMatch(SingleNodeUpdateResponse::isSuccessful)
```

Comment:
some of these can still be refactored to be easier to read

Reply:
pulled out a boolean here and checked the other if-statements.